python/cmsis_svd/parser.py

```diff
@@ -243,6 +264,54 @@ def duplicate_array_of_registers(input):    #expects a SVDRegister which is an a
             )
         )
     return output
+
+
+
+def propagate_defaults(device):
+    def inherit_register_defaults(source,dest):
+        if source.size is not None:
+            if dest.size is None:
+                dest.size = source.size
+        if source.access is not None:
+            if dest.access is None:
+                dest.access = source.access
+        if source.protection is not None:
+            if dest.protection is None:
+                dest.protection = source.protection
+        if source.reset_value is not None:
+            if dest.reset_value is None:
+                dest.reset_value = source.reset_value
+        if source.reset_mask is not None:
+            if dest.reset_mask is None:
+                dest.reset_mask = source.reset_mask
+
+
+    if device.peripherals:
+        for peripheral in device.peripherals:
+            if peripheral.derived_from:
+                parent = None;
+                for p in device.peripherals:
+                    if p.name == peripheral.derived_from:
+                        parent = p      #TODO support dot see http://www.keil.com/pack/doc/CMSIS/SVD/html/svd__outline_pg.html
+                if peripheral.description is None:
+                    peripheral.description = parent.description
```

Comment:
As a demonstration of an alternative to this approach, see https://github.com/posborne/cmsis-svd/commit/b277979ac7c572e5f8ef7d8ecada5a25b3a47acf.  The other alternative is to store description in a field like `_description` on the Peripheral and then define a property like the following:

```python
@property
def description(self):
    return self._description or self.device.description
```

Both of those use a clever but convenient form of short-circuit evaluation to filter out the None case.